Makefile

```diff
@@ -83,18 +82,6 @@ check: check-bash vet
 check-bash:
 	find . -name \*.sh -exec bash -n \{\} \;
 
-vagrant-sync-config:
```

Comment:
can we keep the makefile targets and just rename them something generic?

I'd prefer people actually used makefile targets instead of the scripts directly. This way we can reorganize the source tree without breaking everyone's work flows.... in a weird way this is similar to someone using dns instead of IPs directly.  We can update the dns entry with new a new ip and no one has to know as long as they referenced the service using dns. 

maybe use these target names?

cluster-up
cluster-down
cluster-build
cluster-build-optional
cluster-deploy

Reply:
My aim was to remove everything from the makefiles which is not go-code (build + test), formatting or code generation related. The cluster/up.sh, ... scripts would have been the well known entry-points for the cluster functionality. What I wanted to achieve is, to make it easy to understand that we can just wrap every makefile call with a build docker container and it will still work.

However, I don't have a too strong opinion about that. I can also adopt the generic makefile targets.

Reply:
Yeah, I understand where you're coming from.

My thoughts about keeping the Makefile targets is that it gives us a single entry point for building, deploying, and testing. The analogy I'm making is... Makefile targets are to dns entries as the cluster sync/build/deployment scripts are to IPs.  

If everyone's dev flows and all our automation scripts are mapped to executing makefile targets, then we have the freedom to modify the underlying implementation of those targets without worrying about it impacting anything. 

Besides all that, I kind of like the way we utilize the makefile target dependencies for the vagrant deploy.  Deploy involves syncing, building, and cycling the manifests.  But each one of those steps are individual targets that build off one another.

There are no rules about what Phony Targets should be used for.  I've seen makefile targets do all kinds of crazy stuff. I think we had them executing things like Terraform templates at some job I worked at previously.

Reply:
done